drain3/template_miner.py

```diff
@@ -170,24 +174,119 @@ def match(self, log_message: str, full_search_strategy="never") -> LogCluster:
         matched_cluster = self.drain.match(masked_content, full_search_strategy)
         return matched_cluster
 
-    def get_parameter_list(self, log_template: str, content: str):
-        escaped_prefix = re.escape(self.config.mask_prefix)
-        escaped_suffix = re.escape(self.config.mask_suffix)
-        template_regex = re.sub(escaped_prefix + r".+?" + escaped_suffix, self.drain.param_str, log_template)
-        if self.drain.param_str not in template_regex:
-            return []
-        template_regex = re.escape(template_regex)
-        template_regex = re.sub(r'\\ +', r'\\s+', template_regex)
-        template_regex = "^" + template_regex.replace(escaped_prefix + r"\*" + escaped_suffix, "(.*?)") + "$"
+    def get_parameter_list(self, log_template: str, log_message: str) -> List[str]:
+        """
+        Extract parameters from a log message according to a provided template that was generated
+        by calling `add_log_message()`.
 
-        for delimiter in self.config.drain_extra_delimiters:
-            content = re.sub(delimiter, ' ', content)
-        parameter_list = re.findall(template_regex, content)
-        parameter_list = parameter_list[0] if parameter_list else ()
-        parameter_list = list(parameter_list) if isinstance(parameter_list, tuple) else [parameter_list]
+        Is is preferable to use extract_parameters.
 
-        def is_mask(p: str):
-            return p.startswith(self.config.mask_prefix) and p.endswith(self.config.mask_suffix)
+        :param log_template: log template corresponding to the log message
+        :param log_message: log message to extract parameters from
+        :return: An ordered list of parameter values present in the log message.
+        """
 
-        parameter_list = [p for p in list(parameter_list) if not is_mask(p)]
-        return parameter_list
```

Comment:
`warnings.warn("use extract_parameters instead", DeprecationWarning)` could be used here to formalize the deprecation.

Reply:
Please do